site/public/acmApp/components/officers/officersService.js

```diff
@@ -0,0 +1,16 @@
+angular
+  .module('acmApp')
+  .factory('officersService', ['$http', officersService]);
+
+function officersService ($http) {
+  var officersEndpoint = '/officers';
+  var officersService = {
+    getOfficers: getOfficers
+  };
+
+  function getOfficers() {
+    return $http.get(officersEndpoint);
```

Comment:
I think it's better practice to put all the logic in here. Something like:

```
function getOfficers() {
    var deferred = $q.defer();
    $http.get(officersEndpoint)
        .success(function (data) {
            deferred.resolve(data);
        }
        .error(function (err) {
            deferred.reject(err);
        }
    return deferred.promise;
}
```

And then in the controller, you call something like this:

```
officersService.getOfficers().then(function (result) {
     vm.allOfficers = result;
});
```